### PR TITLE
dt-bindings: sensor: add macros for DT property setting

### PR DESCRIPTION
**Pull request by avisconti**
Add  macros for DT properties (such as odr, range, ...) settings. This is helpful to have a more readable .dts/.dtsi file.
For example, in a dts declaring lsm6dsv16x sensor, it is possible to set properties using a macro with a self-explanatory name:

```
    #include <zephyr/dt-bindings/sensor/st_lsm6dsv16x.h>

    lsm6dsv16x: lsm6dsv16x@0 {
      ...

      accel-range = <LSM6DSV16X_DT_FS_8G>;
      accel-odr = <LSM6DSV16X_DT_ODR_AT_60Hz>;
      gyro-range = <LSM6DSV16X_DT_FS_4000DPS>;
      gyro-odr = <LSM6DSV16X_DT_ODR_AT_60Hz>;
   };
```

Impacted sensors:

  - iis2iclx
  - lis2dh
  - iis2dlpc
  - ism330dhcx
  - lis2dw12
  - lis2ds12
  - lps22df
  - lps22hh
  - lsm6dso16is
  - lsm6dso
  - lsm6dsv16x

This PR has one separate commit for each sensor.

### CODE REVIEW

File: dts/bindings/sensor/st,ism330dhcx-common.yaml [hunk 1]
```diff
@@ -38,47 +52,31 @@ properties:
...
+      - 2 # ISM330DHCX_DT_ODR_26H
+      - 3 # ISM330DHCX_DT_ODR_52Hz
+      - 4 # ISM330DHCX_DT_ODR_104Hz
+      - 5 # ISM330DHCX_DT_ODR_208Hz
+      - 6 # ISM330DHCX_DT_ODR_416Hz
+      - 7 # ISM330DHCX_DT_ODR_833Hz
+      - 8 # ISM330DHCX_DT_ODR_1666Hz
+      - 9 # ISM330DHCX_DT_ODR_3332Hz
+      - 10 # ISM330DHCX_DT_ODR_6667Hz
+      - 11 # ISM330DHCX_DT_ODR_1Hz6
```
Comment (erwango) [inline dts/bindings/sensor/st,ism330dhcx-common.yaml:67]: Please fill description field as well. Comments are not rendered in https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/st%2Clis2dh12-i2c.html#dtbinding-st-lis2dh12-i2c

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/st,ism330dhcx-common.yaml (hunk 1 above)
Comment (avisconti) [reply dts/bindings/sensor/st,ism330dhcx-common.yaml:67]: ok sure, my bad...

Comment (avisconti) [reply dts/bindings/sensor/st,ism330dhcx-common.yaml:67]: I'm doing as you are propsing. Nevertheless if I see following doc

```https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/st,ism330dhcx-i2c.html```

the gyro-range property should not have the correct description, but it has.
Am I missing something?

File: include/zephyr/dt-bindings/sensor/st_lsm6dsv16x.h
[the remark this answers is not in the document]
Comment (avisconti) [reply include/zephyr/dt-bindings/sensor/st_lsm6dsv16x.h:1]: sounds good

**Review by fabiobaltieri — DISMISSED** [review]
Nice job, worth an entry in the migration guide under "Recommended Changes"? Just a line mentioning that these are available and pointing at this PR is what I'm thinking.

**Comment by avisconti** [comment]
> Nice job, worth an entry in the migration guide under "Recommended Changes"? Just a line mentioning that these are available and pointing at this PR is what I'm thinking.

But should I add the commit to this PR itself?

**Comment by fabiobaltieri** [comment]
> But should I add the commit to this PR itself?

Yeah I'd say so.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by teburd — APPROVED** [review]
I like this quite a lot I have to say

**Comment by avisconti** [comment]
> I like this quite a lot I have to say

Thank you!

**Review by erwango — APPROVED** [review]
Nice!

**Pull request merged**